backend/services/implementations/testSessionService.ts

```diff
@@ -187,6 +204,78 @@ class TestSessionService implements ITestSessionService {
 
     return testSessionDtos;
   }
+
+  /*
+   * computeTestGrades computes the breakdown and score of a given result and returns the ResultDTO
+   */
+  async computeTestGrades(
+    result: NewResultDTO,
+    testId: string,
+  ): Promise<ResultDTO> {
+    let resultDto: ResultDTO;
+    let questionsCorrect = 0; // the number of questions the student gets correct
+    const studentAnswers: number[] = result.answers; // the array of the student's raw answers
+    let updatedScore = 0.0; // the final score that will be returned
```

Comment:
nit: let's call this `computedScore` or something, since I don't think we have a score before, so don't think we are updating anything?